inc/itilfollowup.class.php

```diff
@@ -1123,73 +1119,7 @@ public static function buildParentCondition(
          $condition = "(`itemtype` = '$itemtype' AND (0 = 1 ";
-
-         if (Session::haveRight($itemtype, $itemtype::READMY)) {
-            // Add tickets where the users is requester, observer or recipient
-            // Subquery for requester/observer user
-            $user_query = "SELECT `$target`
-               FROM `$user_table`
-               WHERE `users_id` = '$user' AND type IN ($requester, $obs)";
-            $condition .= "OR `items_id` IN ($user_query) ";
-
-            // Subquery for recipient
-            $recipient_query = "SELECT `id`
-               FROM `$table`
-               WHERE `users_id_recipient` = '$user'";
-            $condition .= "OR `items_id` IN ($recipient_query) ";
-         }
-
-         if (Session::haveRight($itemtype, $itemtype::READGROUP)) {
-            // Add tickets where the users is in a requester or observer group
-            // Subquery for requester/observer group
-            $group_query = "SELECT `$target`
-               FROM `$group_table`
-               WHERE `groups_id` IN ($groups) AND type IN ($requester, $obs)";
-            $condition .= "OR `items_id` IN ($group_query) ";
-         }
-
-         if (Session::haveRightsOr($itemtype, [
-            $itemtype::OWN,
-            $itemtype::READASSIGN
-         ])) {
-            // Add tickets where the users is assigned
-            // Subquery for assigned user
-            $user_query = "SELECT `$target`
-               FROM `$user_table`
-               WHERE `users_id` = '$user' AND type = $assign";
-            $condition .= "OR `items_id` IN ($user_query) ";
-         }
-
-         if (Session::haveRight($itemtype, $itemtype::READASSIGN)) {
-            // Add tickets where the users is part of an assigned group
-            // Subquery for assigned group
-            $group_query = "SELECT `$target`
-               FROM `$group_table`
-               WHERE `groups_id` IN ($groups) AND type = $assign";
-            $condition .= "OR `items_id` IN ($group_query) ";
-
-            if (Session::haveRight('ticket', Ticket::ASSIGN)) {
-               // Add new tickets
-               $tickets_query = "SELECT `id`
-                  FROM `$table`
-                  WHERE `status` = '" . CommonITILObject::INCOMING . "'";
-               $condition .= "OR `items_id` IN ($tickets_query) ";
-            }
-         }
-
-         if (Session::haveRightsOr('ticketvalidation', [
-            TicketValidation::VALIDATEINCIDENT,
-            TicketValidation::VALIDATEREQUEST
-         ])) {
-            // Add tickets where the users is the validator
-            // Subquery for validator
-            $validation_query = "SELECT `$target`
-               FROM `glpi_ticketvalidations`
-               WHERE `users_id_validate` = '$user'";
-            $condition .= "OR `items_id` IN ($validation_query) ";
-         }
-
-         return $condition .= ")) ";
+         return $condition . Ticket::buildCanViewCondition("items_id") . ")) ";
```

Comment:
Does not seems correct to use `Ticket` here

Reply:
What would you use instead ?
This is building the condition for ITILFollowups whose parents are Tickets.
The new code for TicketTask needed to use it too (every TicketTask have a parent Ticket) so it seemed right to export in into Ticket.

Reply:
"What would you use instead ?" => `$itemtype`;)

Reply:
This line is inside this condition: 
```php
if ($itemtype == "Ticket")
```
So $itemtype can only be "Ticket".

I think it's better to use the string rather than $itemtype as it makes the code easier to read and may enable some features of your IDE like "Find references" or "Go to definition" if supported.

Reply:
if it's only for TicketFollowup, inherits buildParentCondition in this class,  call parent for need code and apply specific behavior.
We should not have the kind of condition mentioned above

Reply:
nevermind, `TicketFollowup` is deprecated.